### PR TITLE
Existence/Throw/Expansion AST

### PR DESCRIPTION
**Pull request by helixbass**
@GeoffreyBooth PR for ASTs for Existence, Throw and Expansion nodes

Sorry to have fallen off the pace. Was getting crushed on a two-week solo project. But through with that, and finished getting `ast` (ie the new AST methods) merged into my branch (with Prettier plugin and ESLint plugin tests passing against the updated branch) so set up for smooth sailing 👍 

That being said, these are the last of the "easy" leaf node type ASTs. I think it makes sense next to introduce block ASTs so that we can start getting into functions/control structures/etc

### CODE REVIEW

File: src/nodes.coffee [hunk 1]
```diff
@@ -3444,6 +3444,11 @@ exports.Expansion = class Expansion extends Base
 
   eachName: (iterator) ->
 
+  astType: -> 'RestElement'
```
Comment (helixbass) [inline src/nodes.coffee]: While `Expansion`s are Coffeescript-specific, it makes sense to model them like their close cousins splats, so `type: 'RestElement'` just with a `null` `argument` field

File: src/nodes.coffee [hunk 2]
```diff
@@ -3911,6 +3921,13 @@ exports.Existence = class Existence extends Base
       code = "#{code} #{cmp} #{@comparisonTarget}"
     [@makeCode(if o.level <= LEVEL_COND then code else "(#{code})")]
 
+  astType: -> 'UnaryExpression'
```
Comment (helixbass) [inline src/nodes.coffee]: Initially I'd exposed `Existence` as its own unique Coffeescript-specific `type: 'Existence'`

But then realized that conceptually it's a `UnaryExpression` (like eg `!a`), just with a postfix operator. `UnaryExpression` ASTs already have a `prefix` field, so this doesn't even require extending the Babel-style AST at all

File: src/nodes.coffee (hunk 1 above)
Comment (GeoffreyBooth) [reply src/nodes.coffee]: Unless `RestElement` with a `null` `argument` might someday be used for something else in the Babel spec?

Comment (helixbass) [reply src/nodes.coffee]: That's an interesting consideration. Conceptually I think this representation makes sense (an expansion basically is a "spread without an argument") and I'd tend not to worry too hard about potential future evolution of the Babel spec (as we've already seen, it's a bit of a moving target)

File: src/nodes.coffee (hunk 2 above)
Comment (helixbass) [reply src/nodes.coffee]: Sorry I didn't mean literally the same as `!a`, just using `!` as an example where the general semantics is similar to `Existence` (a unary operator whose output is a boolean based on whether its argument is some form of "truthy" or not)

So to be clear, the AST for `!a` looks like
```
type: 'UnaryExpression'
operator: '!'
argument:
  type: 'Identifier'
  name: 'a'
prefix: yes
```
whereas the AST for `a?` looks like
```
type: 'UnaryExpression'
operator: '?'
argument:
  type: 'Identifier'
  name: 'a'
prefix: no
```

Comment (GeoffreyBooth) [reply src/nodes.coffee]: That makes much more sense. Sounds good 👍

**Pull request merged**